pay-api/src/pay_api/services/eft_service.py

```diff
@@ -121,14 +126,39 @@ def create_shortname_refund(cls, request: Dict[str, str], **kwargs) -> Dict[str,
         current_app.logger.debug(f'Starting shortname refund : {shortname_id}')
 
         refund = cls._create_refund_model(request, shortname_id, amount, comment)
-        recipients = EFTRefundEmailList.find_all_emails()
+        cls._refund_eft_credits(int(shortname_id), amount)
 
+        recipients = EFTRefundEmailList.find_all_emails()
         subject = f'Pending Refund Request for Short Name {shortname}'
         html_body = cls._render_email_body(shortname, amount, comment, shortname_id)
 
         send_email(recipients, subject, html_body, **kwargs)
         refund.save()
 
+    @classmethod
+    def _refund_eft_credits(cls, shortname_id: int, amount: str):
+        """Refund the amount to eft_credits table based on short_name_id."""
+        refund_amount = Decimal(amount)
+        eft_credits: List[EFTCreditModel] = EFTShortnames.get_eft_credits(shortname_id)
+        eft_credit_balance = EFTShortnames.get_eft_credit_balance(shortname_id)
+
+        if refund_amount > eft_credit_balance:
+            current_app.logger.error(f'Shortname {shortname_id} Refund amount exceed eft_credits remaining amount.')
+            raise BusinessException(Error.INVALID_TRANSACTION)
```

Comment:
I personally would add a new Error for this case, but I will leave it up to you.

Do we need the logging if we are already raising an exception?